README.md

```diff
@@ -252,6 +255,12 @@ with dependencies between tests (or tests that write to shared locations). When
 enabling `--jobs` for the first time be sure to re-run bats multiple times to
 identify any inter-test dependencies or non-deterministic test behaviour.
 
+If your code relies on variables from the environment, or from `setup_file()`, 
+you need to specify `--parallel-preserve-environment` as well. Note that this 
+requires running `parallel --record-env` first as a setup step as GNU Parallel 
```

Comment:
Is this something we could automate?

Reply:
I am afraid that a fully automatic detection would amount to executing the code to see what variables it does reference. A simpler solution could be to have a `BATS_PRESERVE_SETUP_FILE_VARIABLES` variable that takes all variable that should be preserved. This is more explicit and gives users the opportunity to make up their own mind instead of taking all variables.

Then there would be the additional requirement of environment variables of the bats run that we might want to forward but that could be solved in the same way.

Reply:
I just noticed that you referred to `parallel --record-env`. There will be an explicit error message if this has not been run before. I opted for explicit user action because the results might be non obvious: this command records the current environment as to ignore. This could result in unintentionally ignoring variables that the user explicitly wants to forward.